pkg/middleware/recovery/recovery_test.go

```diff
@@ -0,0 +1,158 @@
+					grpc_recovery.WithRecoveryHandlerContext(
+						PanicRecoveryHandler(logger.MustNewLogger("text", "info", "unix")),
+					),
+				),
+			}...,
+		),
+	}
+
+	t.Run("With_Unary_Panic_Interceptor", func(t *testing.T) {
+		srv := grpc.NewServer(serverOpts...)
+		t.Cleanup(srv.Stop)
+
+		openfgav1.RegisterOpenFGAServiceServer(srv, &unimplementedOpenFGAServiceServer{})
+
+		go func() {
+			err := srv.Serve(listner)
+			if err != nil {
+				t.Fail()
+			}
+		}()
+
+		dialer := func(context.Context, string) (net.Conn, error) {
+			return listner.Dial()
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+		t.Cleanup(cancel)
+
+		opts := []grpc.DialOption{
+			grpc.WithContextDialer(dialer),
+			grpc.WithTransportCredentials(insecure.NewCredentials())}
+
+		conn, err := grpc.DialContext(ctx, "", opts...)
+		require.NoError(t, err)
+
+		t.Cleanup(func() {
+			conn.Close()
+		})
+
+		cli := openfgav1.NewOpenFGAServiceClient(conn)
+
+		_, err = cli.Check(ctx, &openfgav1.CheckRequest{})
+		require.ErrorContains(t, err, http.StatusText(http.StatusInternalServerError))
+	})
+}
+
+func TestStreamPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
+	t.Cleanup(func() {
+		listner.Close()
+	})
+
+	serverOpts := []grpc.ServerOption{
+		grpc.ChainStreamInterceptor(
+			[]grpc.StreamServerInterceptor{
+				grpc_recovery.StreamServerInterceptor(
+					grpc_recovery.WithRecoveryHandlerContext(
+						PanicRecoveryHandler(logger.MustNewLogger("text", "info", "unix")),
+					),
+				)}...,
+		),
+	}
+
+	t.Run("With_Stream_Panic_Interceptor", func(t *testing.T) {
+		srv := grpc.NewServer(serverOpts...)
+		t.Cleanup(srv.Stop)
+
+		openfgav1.RegisterOpenFGAServiceServer(srv, &unimplementedOpenFGAServiceServer{})
+
+		go func() {
+			_ = srv.Serve(listner)
+		}()
+
+		dialer := func(context.Context, string) (net.Conn, error) {
+			return listner.Dial()
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+		t.Cleanup(cancel)
+
+		opts := []grpc.DialOption{
+			grpc.WithContextDialer(dialer),
+			grpc.WithTransportCredentials(insecure.NewCredentials())}
+
+		conn, err := grpc.DialContext(ctx, "", opts...)
+		require.NoError(t, err)
+
+		cli := openfgav1.NewOpenFGAServiceClient(conn)
+		stream, err := cli.StreamedListObjects(ctx, &openfgav1.StreamedListObjectsRequest{})
+		require.NoError(t, err)
+
+		_, err = stream.Recv()
+		require.ErrorContains(t, err, http.StatusText(http.StatusInternalServerError))
+	})
+}
+
+type unimplementedOpenFGAServiceServer struct {
+	openfgav1.UnimplementedOpenFGAServiceServer
+}
+
+func (unimplementedOpenFGAServiceServer) Check(context.Context, *openfgav1.CheckRequest) (*openfgav1.CheckResponse, error) {
+	panic("Unexpected error!")
+}
+
+func (unimplementedOpenFGAServiceServer) StreamedListObjects(m *openfgav1.StreamedListObjectsRequest, stream openfgav1.OpenFGAService_StreamedListObjectsServer) error {
+	_ = stream.RecvMsg(m)
+
+	panic("Unexpected error!")
+}
```

Comment:
The tests can be made simpler using the testing utilities provided by https://github.com/grpc-ecosystem/go-grpc-middleware/blob/main/testing/testpb/interceptor_suite.go#L35

```go
var pingReq = &testpb.PingRequest{Value: "ping"}

type pingService struct {
	testpb.TestServiceServer
	T *testing.T
}

func (s *pingService) Ping(ctx context.Context, req *testpb.PingRequest) (*testpb.PingResponse, error) {
	panic("boom!")
	return s.TestServiceServer.Ping(ctx, req)
}

func (s *pingService) PingStream(ss testpb.TestService_PingStreamServer) error {
	panic("boom!")
	return s.TestServiceServer.PingStream(ss)
}

func TestRecoveryMiddleware(t *testing.T) {
	s := &PanicRecoveryTestSuite{
		InterceptorTestSuite: &testpb.InterceptorTestSuite{
			TestService: &pingService{&testpb.TestPingService{}, t},
			ServerOpts: []grpc.ServerOption{
				grpc.UnaryInterceptor(grpc_recovery.UnaryServerInterceptor(
					grpc_recovery.WithRecoveryHandlerContext(
						PanicRecoveryHandler(logger.MustNewLogger("text", "info", "unix")),
					),
				)),
				grpc.StreamInterceptor(
					grpc_recovery.StreamServerInterceptor(
						grpc_recovery.WithRecoveryHandlerContext(
							PanicRecoveryHandler(logger.MustNewLogger("text", "info", "unix")),
						),
					)),
			},
		},
	}

	suite.Run(t, s)
}

type PanicRecoveryTestSuite struct {
	*testpb.InterceptorTestSuite
}

func (s *PanicRecoveryTestSuite) TestPing() {
	_, err := s.Client.Ping(s.SimpleCtx(), pingReq)
	s.Require().ErrorContains(err, "Internal Server Error")
}

func (s *PanicRecoveryTestSuite) TestStreamingPing() {
	stream, err := s.Client.PingStream(s.SimpleCtx())
	_, err = stream.Recv()
	s.Require().ErrorContains(err, "Internal Server Error")
}
```

Reply:
There are examples where we do this in the `requestid` middleware, for exmaple.

Reply:
@smeadows-okta is there a reason why you dismissed this suggestion? I.e. is the snippet I provided worse than your current implementation?

Reply:
I thought Jon answered. gRPC ping is used for service-to-service gRPC requests where service is not available. My thoughts are, we already have service openFGA for end-to-end testing and where testing should be conducted. The code snippet is great and can be implemented, I have no issue with either direction.

Reply:
> gRPC ping is used for service-to-service gRPC requests where no service is available

I didn't understand this. The Ping Service is just a proxy to test the actual middleware.

In any case, it's fine, you spent the time implementing your version, I don't want to let that effort go to waste.